test/CMakeLists.txt

```diff
@@ -1,9 +1,18 @@
-cmake_minimum_required(VERSION 3.16)
+cmake_minimum_required(VERSION 3.24)
 
 project(concurrencppTests LANGUAGES CXX)
 
 include(../cmake/coroutineOptions.cmake)
 
+if(NOT DEFINED CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+  get_cmake_property(GENERATOR_IS_MULTI_CONFIG GENERATOR_IS_MULTI_CONFIG)
```

Comment:
will this work with the current way to build tests?
`cmake -S test -B build/test ;
cmake --build build/test`